akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -699,6 +703,16 @@ private[akka] class ShardRegion(
     retryCount = 0
   }
 
+  def deliverStartEntity(msg: StartEntity, snd: ActorRef): Unit = {
+    try {
+      deliverMessage(msg, snd)
+    } catch {
+      case ex: MatchError ⇒
+        log.error(ex, "When using remember-entities the shard id extractor must handle ShardRegion.StartEntity(id).")
+      // TODO fail sharding entirely?
```

Comment:
But it does mean that remember entities does not work at all when someone has it enabled, isn't that a valid reason to fail shard startup (if possible)?